hal/ia32/interrupts.c

```diff
@@ -122,24 +122,28 @@ int interrupts_dispatchIRQ(unsigned int n, cpu_context_t *ctx)
 	int reschedule = 0;
 	spinlock_ctx_t sc;
 
-	if (n >= SIZE_INTERRUPTS)
+	if (n >= SIZE_INTERRUPTS) {
 		return 0;
+	}
 
 	hal_spinlockSet(&interrupts.spinlocks[n], &sc);
 
 	interrupts.counters[n]++;
 
 	if ((h = interrupts.handlers[n]) != NULL) {
```

Comment:
regarding changes in the MISRA commit
```suggestion
	h = interrupts.handlers[n];
	if (h != NULL) {
```